File: Level 2/[3차] n진수 게임/solution.js
function solution(n, t, m, p) {
  let string = "";
  let num = 0;

  while (string.length < m * t) {
    string += num.toString(n);
    num++;
  }

  let answer = "";

  for (let i = p - 1; i < string.length; i += m) {
    answer += string[i];
    if (answer.length === t) {
      return answer.toUpperCase();
    }
  }
}

File: Level 2/[3차] n진수 게임/solution.py
def solution(n, t, m, p):
    string = ""
    num = 0

    while len(string) < m * t:
        string += convert(num, n)
        num += 1

    return string[p - 1 :: m][:t]


def convert(num, n):
    temp = "0123456789ABCDEF"
    div, mod = divmod(num, n)

    if div == 0:
        return temp[mod]
    else:
        return convert(div, n) + temp[mod]
